File: API_Captura_Indiv/API_Captura-main/power-tech/CodigoPythonConst.py
import psutil
import time
import mysql.connector

UsoCpu = psutil.cpu_percent(interval=1)
try:
    mydb = mysql.connector.connect(host = '3.234.2.175', user = 'root',password = 'urubu100',database = 'PowerTechSolutions')
    if mydb.is_connected():
        db_info = mydb.get_server_info()
        mycursor = mydb.cursor()
        sql_querryCPU = 'INSERT INTO Monitoramento_RAW VALUES (NULL, CURRENT_TIMESTAMP(),NULL, NULL, %s, NULL, 1)'
        dado = (round(UsoCpu,2),)
        mycursor.execute(sql_querryCPU, dado)
        mydb.commit()
finally:
    if mydb.is_connected():
        mycursor.close()
        mydb.close()

File: API_Captura_Indiv/API_Captura-main/power-tech/src/main/kotlin/CodigoPythonConst.kt
import java.io.File

object CodigoPythonConst {

    fun execpython(servicos:MutableList<ServicosMonitorados>) {

        val servicoCadastradorepositorio = ServicoCadastradoRepositorio()
        servicoCadastradorepositorio.iniciarSql()

        var CPU = 0
        var componenteCPU = 0

        for (servico in servicos){

            var apelido = servicoCadastradorepositorio.buscarComponente(servico.FKComponente_cadastrado)

            when(apelido){
                "CPU" -> {
                    CPU += 1
                    componenteCPU = servico.IDComponente_monitorado
                }

            }

        }


        var codigoPython ="""
import psutil
import time
import mysql.connector
import pyodbc
        
UsoCpu = psutil.cpu_percent(interval=1)
        
conn_mssql = pyodbc.connect(
    'DRIVER=ODBC DRIVER 17 for SQL server;'
    'Server=ec2-34-194-127-191.compute-1.amazonaws.com;'
    'UID=sa;'
    'PWD=myLOVEisthe0506;'
    'Database=PowerTechSolutions;'
)
cursor_mssql = conn_mssql.cursor()    
componenteCPU = $componenteCPU
dado = (round(UsoCpu, 2), componenteCPU)
sql_insert_CPU = f"INSERT INTO Monitoramento_RAW (Uso,FKComponente_Monitorado) VALUES ({round(UsoCpu, 2)},{componenteCPU})"
    
cursor_mssql.execute(sql_insert_CPU)
conn_mssql.commit()
"""

        val nomeArquivoPyDefault = "CodigoPythonConstErica.py"

        File(nomeArquivoPyDefault).writeText(codigoPython)
        Runtime.getRuntime().exec("python3 $nomeArquivoPyDefault")

        println("Python executado para CPU")

    }

}